docs/prover/diagnosis/index.md

```diff
@@ -5,15 +5,238 @@ Diagnostic Tools
+nodes have a dashed outline. Clicking on a regular nodes will make the source
+code box (discussed below) focus on the corresponding source code; clicking on a
+call node will replace the currently displayed CFG with the CFG that belongs to
+the called method. 
+
+```{note}
+Only external calls are explicit in the TAC report's CFGs. Internal calls are 
+inlined on the TAC source code level.
+```
+
+[^nested-cfg]: Strictly speaking, there is a set of CFGs available for each
+    verification condition. Every external call has its own CFG, and the CFGs
+    are related by call nodes which lead from a call site to the corresponding
+    callee's CFG. Intuitively, this set can be viewed as one CFG with nested 
+    sub-CFGs for the calls.
+
+The upper-mid left part of a TAC report contains the TAC source code. The
+details of the TAC language are outside the scope of this documentation.
+Generally speaking, TAC is generated by compiler-style transformations of the
+EVM bytecode together with the CVL specification. Many patterns from EVM
+bytecode are retained in TAC, however inlining of functions and various
+simplifications have already happened. Especially since the source code in the
+TAC reports is taken from a stage directly before the SMT solving. One benefit
+of viewing this stage of the code is that it has all the summarizations,
+inlinings and optimizations already applied, it is very close to what the SMT
+solver sees. 
+
+Some commands in the TAC source code have pointers to the program or
+specification code that the TAC program was created from. These commands are
+highlighted a dotted underline; the source code location they have been
+translated from is revealed on hovering with the mouse over such a command, as
+displayed in the image below.
+
+
+```{figure} source-pointer-on-hover-annotated.png
+:name: source-pointer-on-hover
+:alt: Source pointer shown on hover
+:align: center
+
+source pointer is shown on hover over TAC commands that have a dotted underline
+```
+
+The call-based navigation on the mid-right side of the report lists all the
+calls in the TAC program. Each call is identified by a number. After the number
+there is a number of arrows, indicating the nesting depth of the call, i.e.
+calls from spec get one arrow, a call from a method body with one arrow gets two
+arrows, etc.; see the picture below for an example. After the arrows the name of
+the called method is given. Each of these calls is clickable and will lead to
+the corresponding CFG.
+
+```{figure} call-depth-arrows.png
+:name: call-depth-arrows
+:alt: Arrows indicating depth of call
+:align: center
+:height: 300
+
+arrows indicating the "depth" of a call in the call links
+```
+
+% TODO: ok to show these Aave/delvtech|element method names in this picture?
+
+At the bottom of the TAC report are the successor and predecessor relations of
+the CFG on a per-block/per-node basis. These are usually not useful for the tool
```

Comment:
`a per-block/per-node basis`. The reader does not know what a `block` is. Also, it does not show a relation on a `per-node` basis (not in the sense of `nodes` as you described above). I would just say something like

"The bottom part of the TAC report shows a more-detailed control-flow relations within individual nodes. These are usually not useful for the tool user, ..."